backend/de.metas.manufacturing/src/main/java/org/eevolution/model/MPPOrder.java

```diff
@@ -286,6 +288,11 @@ public String completeIt()
 	@Override
 	public boolean voidIt()
 	{
+		final PPOrderCandidateService ppOrderCandidateService = SpringContextHolder.instance.getBean(PPOrderCandidateService.class);
+		final PPOrderChangedEventFactory eventFactory = PPOrderChangedEventFactory.newWithPPOrderBeforeChange(
```

Comment:
refactor and introduce method 
PPOrderChangedEventFactory newPPOrderChangedEventFactory()

(use it in closeIt too)

Reply:
done.